main/realmain.cc

```diff
@@ -589,9 +589,14 @@ int realmain(int argc, char *argv[]) {
             if (gs->hadCriticalError()) {
                 gs->errorQueue->flushAllErrors(*gs);
             }
-            indexed = move(pipeline::typecheck(gs, move(indexed), opts, *workers).result());
-            if (gs->hadCriticalError()) {
-                gs->errorQueue->flushAllErrors(*gs);
+
+            // pipeline::resolve runs the namer, and will exit early if `--stop-after namer` is present.
+            // pipeline::typecheck doesn't need to be run if that flag was present.
+            if (opts.stopAfterPhase != options::Phase::NAMER) {
+                indexed = move(pipeline::typecheck(gs, move(indexed), opts, *workers).result());
+                if (gs->hadCriticalError()) {
+                    gs->errorQueue->flushAllErrors(*gs);
+                }
```

Comment:
https://github.com/sorbet/sorbet/blob/a73ce091fe2ee639a5b970fd994a47b5f8089879/main/pipeline/pipeline.cc#L549

Should we delete this?

Reply:
I think so. Thanks for this comment, I think moving the namer case above the trailing end of the resolver here is better 👍